### PR TITLE
SNOW-1464131: Add client side stored proc tests

### PR DESCRIPTION
**Pull request by sfc-gh-lmukhopadhyay**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1464131

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Adding basic stored procedure tests

### CODE REVIEW

File: tests/integ/modin/test_modin_stored_procedures.py [hunk 1]
```diff
@@ -0,0 +1,389 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import pytest
+
+try:
```
Comment (sfc-gh-azhan) [inline tests/integ/modin/test_modin_stored_procedures.py:8]: why we need this?

File: tests/integ/modin/test_modin_stored_procedures.py [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+
+pytestmark = [
+    pytest.mark.udf,
+]
+
+tmp_stage_name = Utils.random_stage_name()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup(session, resources_path, local_testing_mode):
```
Comment (sfc-gh-azhan) [inline tests/integ/modin/test_modin_stored_procedures.py:28]: why we need this?

File: tests/integ/modin/test_modin_stored_procedures.py [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+    Utils.upload_to_stage(
+        session, tmp_stage_name, test_files.test_sp_py_file, compress=False
+    )
+
+
+@sql_count_checker(no_check=True)
+def test_sproc_head(session):
+    @sproc(
+        packages=[
+            "pandas==2.2.1",
```
Comment (sfc-gh-azhan) [inline tests/integ/modin/test_modin_stored_procedures.py:42]: let's create a constant for those packages.

File: tests/integ/modin/test_modin_stored_procedures.py [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+    test_files = TestFiles(resources_path)
+    if not local_testing_mode:
+        Utils.create_stage(session, tmp_stage_name, is_temporary=True)
+        session.add_packages("snowflake-snowpark-python")
+    Utils.upload_to_stage(
+        session, tmp_stage_name, test_files.test_sp_py_file, compress=False
+    )
+
+
+@sql_count_checker(no_check=True)
```
Comment (sfc-gh-azhan) [inline tests/integ/modin/test_modin_stored_procedures.py:38]: why skip checks?

File: tests/integ/modin/test_modin_stored_procedures.py (hunk 3 above)
Comment (sfc-gh-joshi) [reply tests/integ/modin/test_modin_stored_procedures.py:42]: you should be able to get it from `supported_pandas_version` and `supported_modin_version` in `snowflake/snowpark/modin/plugin/__init__.py`

File: tests/integ/modin/test_modin_stored_procedures.py (hunk 2 above)
Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/test_modin_stored_procedures.py:28]: I'll remove this, it was from snowpark test_stored_procedures but we don't need this local_testing_mode case

File: tests/integ/modin/test_modin_stored_procedures.py (hunk 4 above)
Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/test_modin_stored_procedures.py:38]: I was assuming if tests are run in stored procs, we aren't concerned with sql count checks. Should I add them?

File: tests/integ/modin/test_modin_stored_procedures.py (hunk 3 above)
Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/test_modin_stored_procedures.py:42]: Nice, thanks @sfc-gh-joshi

**Review by sfc-gh-azhan — APPROVED** [review]
Can we add one apply example? Other than that, it is good to go.

File: tests/integ/modin/test_modin_stored_procedures.py [hunk 5]
```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import modin.pandas as pd  # noqa: F401
```
Comment (sfc-gh-evandenberg) [inline tests/integ/modin/test_modin_stored_procedures.py:6]: Why is noqa: F401 needed here?  It looks like pd is used.

**Review by sfc-gh-evandenberg — APPROVED** [review]
LGTM

File: tests/integ/modin/test_modin_stored_procedures.py (hunk 5 above)
Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/test_modin_stored_procedures.py]: True it's not needed, removed

**Pull request merged**